### PR TITLE
[master] Add pretty printer for evm result.

### PR DESCRIPTION
**Pull request by its-saeed**
## Backward Compatibility

- [ ] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [ ] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

**Comment by its-saeed** [comment]
output:
```
apply {
  modify {
    address: f0cb24ac66ba7375bf9b9c4fa91e208d9eaabd2e,
    balance: "0x0000…ff63",
    nonce: "0x0000…0047",
  }
}apply {
  delete {
    address: 3e8cf9e43201989b4cbd1c57759dc53782d08a5f
  }
}

apply {
  modify {
    address: 4c303273f4df4481ff758756498f4bc4266b3316,
    balance: "0x0000…0000",
    nonce: "0x0000…0000",
    storage: [      {
        key: [b"0000000000000000000000000000000000000000000000000000000000000003"],
        value: 0x4d79…000e
      },
      {
        key: [b"0000000000000000000000000000000000000000000000000000000000000004"],
        value: 0x544b…0006
      },
      {
        key: [b"6F59EC273BA59F8ACE412173CEBB96D1CF49AAD9AD073E532F53DAAD4169D3CB"],
        value: 0x0000…0001
      },
      {
        key: [b"A7ACF36970B15453215F8F03B0AF9FAB68405ACA3ECA49E5667D15BE05E8F767"],
        value: 0x0000…0001
      },
    ]
  }
}

```

File: evm-ds/src/precompiles/blake2.rs [hunk 1]
```diff
@@ -200,7 +200,7 @@ fn required_gas(input: &[u8]) -> Result<u64, ExitError> {
 #[cfg(test)]
 mod tests {
     use super::*;
-    type PrecompileResult = Result<PrecompileOutput, PrecompileFailure>;
+    type PrecompileResult = Result<(PrecompileOutput, u64), PrecompileFailure>;
```
Comment (valeryz) [inline evm-ds/src/precompiles/blake2.rs:203]: Why did you have to change this? Is something wrong with the `evm` dependency versioning?

@bzawisto PTAL.

File: evm-ds/src/pretty_printer.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
+use primitive_types::H256;
+use protobuf::Message;
+use log::debug;
+use std::fmt::Write;
+
+use crate::protos::{Evm as EvmProto, ScillaMessage};
+
+fn address_to_string(address: &EvmProto::Address) -> String {
```
Comment (valeryz) [inline evm-ds/src/pretty_printer.rs:8]: I think you could just use the `Into` trait implementations from `convert.rs`, and not duplicate code.

File: evm-ds/src/pretty_printer.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+fn uint256_to_string(number: &EvmProto::UInt256) -> String {
+    let buffer = [number.get_x0().to_be_bytes(), number.get_x1().to_be_bytes(), number.get_x2().to_be_bytes(), number.get_x3().to_be_bytes()].concat();
+    let number_string = H256::from_slice(&buffer);
+    number_string.to_string()
+}
+
+fn apply_modify_to_string(modify: &EvmProto::Apply_Modify) -> String {
+    let mut modify_string = String::new();
+    write!(modify_string, "  modify {{\n    address: {},\n    balance: {:?},\n    nonce: {:?},\n",
+            address_to_string(modify.get_address()), uint256_to_string(modify.get_balance()),
```
Comment (valeryz) [inline evm-ds/src/pretty_printer.rs:25]: `H160::from(modify.get_address())` and then you don't need `address_to_string`

File: evm-ds/src/pretty_printer.rs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            uint256_to_string(modify.get_nonce())).unwrap();
```
Comment (valeryz) [inline evm-ds/src/pretty_printer.rs:26]: Same, `U256::from(....)`

Comment (valeryz) [reply evm-ds/src/pretty_printer.rs:26]: need to use convert for the trait to be in scope.

**Review by valeryz — COMMENTED** [review]
Can you avoid conversions which are impolemented already in `convert.rs`

File: evm-ds/src/precompiles/blake2.rs (hunk 1 above)
Comment (its-saeed) [reply evm-ds/src/precompiles/blake2.rs:203]: @valeryz These changes were in blake2 tests. I think someone had changed the return type but didn't update the tests. They failed to build.

**Review by valeryz — APPROVED** [review]

**Review by valeryz — APPROVED** [review]

**Pull request merged**